ServiceLayer/Services.cs

```diff
@@ -60,12 +60,49 @@ public Result GetBookById(int id)
 
         }
 
-        public bool Update(Book book)
+        public Result Update(Book book)
         {
-            return _BookRepository.ReplaceEmployee(book);
+            Result result;
+            result = Validate(book);
+
+            if (result.ErrorMessage != null)
+                return result;
+            if( _BookRepository.ReplaceEmployee(book))
+            {
```

Comment:
What is Employee in BookRepository

Reply:
before working with book the worked with employee .....i forget to changed it .Now resolved